### PR TITLE
refactor/242 - Overhaul KJFK

### LINKED ISSUE #242: Airport Revival: John F. Kennedy International Airport
John F. Kennedy International Airport (IATA: JFK, ICAO: KJFK, FAA LID: JFK) is a major international airport located near South Ozone Park in the borough of Queens in New York City, United States, 12 miles (20 km) southeast of Lower Manhattan. It is the busiest international air passenger gateway into the United States,  the fifth busiest airport in the United States and the busiest airport in the New York City airport system, handling 58,956,288 passengers in 2016. Over ninety airlines operate out of the airport, with non-stop or direct flights to destinations in all six inhabited continents. The airport features six passenger terminals and four runways. It serves as a hub for American Airlines and Delta Air Lines and is the primary operating base for JetBlue Airways. In the past, JFK served as a hub for Eastern, National, Pan Am, and TWA.

Opened as New York International Airport in 1948, it was commonly known as Idlewild Airport before being renamed in 1963 in memory of John F. Kennedy, the 35th President of the United States, following his assassination.

This airport was omitted when spawnPatterns were introduced.

### PR DESCRIPTION
**Pull request by erikquinn**
Resolves #242.
Replaces #302.

Update KJFK to meet the latest standards and work with the current version of openScope.

### [Test me here!](https://openscope-dev-pr-1460.herokuapp.com/)

### CODE REVIEW

**Comment by erikquinn** [comment]
I think we're done! Let's see what I missed :)

File: assets/airlines/ber.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (Fechulo) [inline assets/airlines/ber.json]: Air Berlin sadly ceased operations on 27 October 2017

File: assets/airlines/ara.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        "long": [
+            ["A332",  2],
+            ["A346",  2]
+        ],
+        "short":[
+            ["CRJ9", 4],
+            ["DH8D", 4]
+        ]
+    }
+}
```
Comment (Fechulo) [inline assets/airlines/ara.json]: Add new line at the end of ara.json

File: assets/airlines/bos.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (Fechulo) [inline assets/airlines/bos.json]: Not sure what happened with the formatting here. Also add new line at the end of the file

File: assets/airlines/cxa.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (Fechulo) [inline assets/airlines/cxa.json]: Add new line at the end of cxa.json

File: assets/airlines/dya.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Fechulo) [inline assets/airlines/dya.json]: Add new line at the end of dya.json

File: assets/airlines/fjm.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Fechulo) [inline assets/airlines/fjm.json]: Add new line at the end of fjm.json

File: assets/airlines/iss.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (Fechulo) [inline assets/airlines/iss.json]: Add new line at the end of iss.json

File: assets/airlines/tai.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+            ["E190",  8 ]
+        ],
+        "long": [
+            ["A332",  1 ]
+        ],
+        "jfk": [
+            ["A320", 18 ]
+        ]
+    }
+}
```
Comment (Fechulo) [inline assets/airlines/tai.json]: Add new line at the end of tai.json

File: assets/airlines/voi.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+        "default": [
+            ["A319", 12],
+            ["A320", 44],
+            ["A321", 10]
+        ],
+        "jfk": [
+            ["A320",  1]
+        ]
+    }
+}
```
Comment (Fechulo) [inline assets/airlines/voi.json]: Add new line at the end of voi.json

File: assets/airlines/xlf.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Fechulo) [inline assets/airlines/xlf.json]: Add new line at the end of xlf.json

File: assets/airports/airportLoadList.js [hunk 11]
```diff
@@ -243,12 +243,12 @@ window.AIRPORT_LOAD_LIST = (function() { // eslint-disable-line wrap-iife
...
-        // {
-        //     icao: 'kjfk',
-        //     level: 'hard',
-        //     name: 'John F Kennedy International Airport',
-        //     premium: false
-        // },
+        {
+            icao: 'kjfk',
+            level: 'hard',
+            name: 'John F Kennedy International Airport',
```
Comment (Fechulo) [inline assets/airports/airportLoadList.js]: Missing the period/full stop/whatever you call it  (`.`) after the F

File: assets/airports/kjfk.json [hunk 12]
```diff
@@ -1,4 +1,5 @@
 {
+    "airac": 1910,
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Line 2: Should be 2001 now, although it'll be 2002 by the time this is merged

File: assets/airports/kjfk.json [hunk 13]
```diff
@@ -7,227 +8,565 @@
     "icao": "KJFK",
     "iata": "JFK",
     "magnetic_north": -13.1,
-    "ctr_radius": 80,
-    "ctr_ceiling": 18000,
+    "ctr_radius": 100,
+    "ctr_ceiling": 17000,
     "initial_alt": 5000,
     "position": ["N40.639926", "W73.778695"],
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Line 14: Add airport elevation

File: assets/airports/kjfk.json [hunk 14]
```diff
@@ -7,227 +8,565 @@
...
+                "KJFK22R": ["#224"],
+                "KJFK31L": ["CRI", ["_CRI176002", "A25+"], "#176"],
+                "KJFK31R": ["CRI", ["_CRI176002", "A25+"], "#176"]
             },
+            "body": ["MERIT"],
             "exitPoints": {
-                "RNGRR": ["SKORR", ["METSS", "A25+"], "RNGRR"],
-                "YNKEE": ["SKORR", ["CESID", "A25+"], "YNKEE"]
+                "MERIT": [],
+                "PUT": ["PUT"]
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Line 419: I believe it goes MERIT then HFD then PUT

File: assets/airports/kjfk.json [hunk 15]
```diff
@@ -7,227 +8,565 @@
...
+                "KJFK13L": ["_RWY31R01DME", "#109"],
+                "KJFK13R": ["_RWY31L01DME", "#109"],
+                "KJFK22L": ["#224"],
+                "KJFK22R": ["#224"],
+                "KJFK31L": ["_RWY13RSKORR057", "SKORR", ["CESID", "A25+"], "YNKEE", "#172"],
+                "KJFK31R": ["_RWY13LSKORR062", "SKORR", ["CESID", "A25+"], "YNKEE", "#172"]
+            },
+            "body": ["DEEZZ", "HEERO"],
+            "exitPoints": {
+                "CANDR": ["CANDR"],
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Line 347: I believe it should be HEERO then KURNL then CANDR

File: assets/airports/kjfk.json [hunk 16]
```diff
@@ -7,227 +8,565 @@
...
+                "HEERO": ["HEERO"],
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Line 348: HEERO is in the body, so this should be empty

File: assets/airports/kjfk.json [hunk 17]
```diff
@@ -7,227 +8,565 @@
...
+                "TOWIN": ["TOWIN"]
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Line 349: I believe it should be HEERO then KURNL then CANDR then TOWIN

File: assets/airports/kjfk.json [hunk 18]
```diff
@@ -7,227 +8,565 @@
...
+            },
+            "draw": [
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Lines 355 to 357 should reflect comments made on lines 347 to 349

File: assets/airports/kjfk.json [hunk 19]
```diff
@@ -7,227 +8,565 @@
...
+        "J230": ["RBV", "COPES", "BYRDD", "SAAME", "LARRI", "BEETS", "VINSE", "LEJOY", "AIR"],
+        "J60": ["SAX", "CANDR", "NEWEL", "GYNTS", "DANNR", "FORTT", "DOOTH", "PSB"],
+        "J75": ["BOS", "SWALO", "NELIE", "GREKI", "CMK", "DUEYS", "JERSY", "SBJ", "BIGGY", "COPES", "MXE", "STOEN", "SACRI", "MURPH", "BINKS", "GVE"],
+        "Q818": ["GAYEL", "MSLIN", "STOMP", "BUFFY", "CFB"]
+    },
     "sids": {
-        "JFK3": {
-            "icao": "JFK3",
-            "name": "KENNEDY THREE",
+        "JFK5": {
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Kennedy 5 departure should be defined below the HAPIE6 departure

File: assets/airports/kjfk.json [hunk 20]
```diff
@@ -7,227 +8,565 @@
...
+              "KJFK22L": [],
+              "KJFK22R": [],
+              "KJFK31L": [],
+              "KJFK31R": []
+            },
+            "draw": [
+                ["LVZ*", "JENNO", "HARTY", "STW", "LENDY", "LGA"]
+            ]
+        },
+        "PARCH2": {
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Line 508: PARCH2 is now PARCH3 (holding distance at PVD was changed)

**Comment by Fechulo** [comment]
@erikquinn It might also be worth adding to the airport guide that departures should be vectored to their departure waypoints, many users won't know that.

File: assets/airlines/ber.json (hunk 1 above)
Comment (erikquinn) [reply assets/airlines/ber.json]: BER removed.

File: assets/airlines/ara.json (hunk 2 above)
Comment (erikquinn) [reply assets/airlines/ara.json]: done

File: assets/airlines/bos.json (hunk 3 above)
Comment (erikquinn) [reply assets/airlines/bos.json]: done. musta been a vscode thing. still getting adjusted... it definitely has its pluses and minuses to say the least :)

File: assets/airlines/cxa.json (hunk 4 above)
Comment (erikquinn) [reply assets/airlines/cxa.json]: done

File: assets/airlines/dya.json (hunk 5 above)
Comment (erikquinn) [reply assets/airlines/dya.json]: done

File: assets/airlines/fjm.json (hunk 6 above)
Comment (erikquinn) [reply assets/airlines/fjm.json]: done

File: assets/airlines/iss.json (hunk 7 above)
Comment (erikquinn) [reply assets/airlines/iss.json]: done

File: assets/airlines/tai.json (hunk 8 above)
Comment (erikquinn) [reply assets/airlines/tai.json]: done

File: assets/airlines/voi.json (hunk 9 above)
Comment (erikquinn) [reply assets/airlines/voi.json]: done

File: assets/airlines/xlf.json (hunk 10 above)
Comment (erikquinn) [reply assets/airlines/xlf.json]: done

File: assets/airports/airportLoadList.js (hunk 11 above)
Comment (erikquinn) [reply assets/airports/airportLoadList.js]: done

File: assets/airports/kjfk.json (hunk 12 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: changed to 2002

File: assets/airports/kjfk.json (hunk 13 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: done

File: assets/airports/kjfk.json (hunk 14 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: huh! good catch... done

File: assets/airports/kjfk.json (hunk 15 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: yep, done

File: assets/airports/kjfk.json (hunk 16 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: done

File: assets/airports/kjfk.json (hunk 17 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: yep, done

File: assets/airports/kjfk.json (hunk 18 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: done

File: assets/airports/kjfk.json (hunk 19 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: done

File: assets/airports/kjfk.json (hunk 20 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: done, looks like all the same 👍

**Comment by erikquinn** [comment]
Added some notes about vectoring departures. A good idea indeed.
> ### Vectoring Departures
>
>With the exception of the SKORR4 departure, RNGRR transition, all departures will need to be vectored to join their route. When departing the 31s, due to airspace constraints (LGA/EWR traffic), all traffic will need a left turn to the west or south.
>
>Aircraft routed over RBV (usually routed `SKORR4.RNGRR..RBV`) are not impacted, but all other aircraft will be issued a southbound heading, climbed above any relevant arrival traffic, and turned toward their departure fix. Departures to north fixes will be climbed above landing traffic, and turned northeasbound across the LENDY arrival flow before being cleared on course.

File: assets/airports/kjfk.json [hunk 21]
```diff
@@ -7,227 +8,623 @@
...
+        "_RWY13LSKORR063": ["N40d39.97m0", "W73d48.52m0"],
+        "_RWY13RSKORR057": ["N40d39.43m0", "W73d50.24m0"],
+        "_RWY13RSKORR058": ["N40d39.38m0", "W73d50.15m0"],
+        "_RWY22L01DME": [40.65954878329, -73.74368000403],
+        "_RWY22R01DME": [40.66484680209, -73.75212018818],
+        "_RWY31L01DME": [40.61947646339, -73.75297764097],
+        "_RWY31R01DME": [40.63520646392, -73.74046224393],
+        "ACK"  : ["N41d16.91m0", "W070d01.60m0"],
+        "AIR"  : ["N40d01.02m0", "W080d49.03m0"],
+        "ALB": ["N42d44.84m0", "W073d48.19m0"],
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Make sure to align the colons on those new fixes 😉

File: assets/airports/kjfk.json [hunk 22]
```diff
@@ -7,227 +8,623 @@
...
+                "KJFK31L": ["_RWY13RSKORR058"],
+                "KJFK31R": ["_RWY13LSKORR063"]
             },
+            "body": ["SKORR"],
             "exitPoints": {
-                "RNGRR": ["SKORR", ["METSS", "A25+"], "RNGRR"],
-                "YNKEE": ["SKORR", ["CESID", "A25+"], "YNKEE"]
+                "SKORR": [],
+                "YNKEE": [["CESID", "A25+"], "YNKEE", "#172"],
+                "RNGRR": [["METSS", "A25+"], "RNGRR", "#223"]
```
Comment (Fechulo) [inline assets/airports/kjfk.json]: Should the `#223` be here?

File: assets/airports/kjfk.json (hunk 21 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: Gah! Forgot 🤦‍♂ 

Fixed now.

File: assets/airports/kjfk.json (hunk 22 above)
Comment (erikquinn) [reply assets/airports/kjfk.json]: Both the FAA and Jeppesen charts are somewhat unclear... to me, I thought the FAA one was inconclusive on whether they meant to show that it was _along_ that radial, or if they meant to fly the radial through RNGRR as the arrow shows. The Jeppesen chart I thought was a little more clearly showing that you should continue along the 223 course, so I went with that. Neither chart had a textual description involving it.

I went back to XPlane (Zibo 737) and actually plugged in `SKORR4 RNGRR RBV J230 COPES J75 GVE` just as an example, and it did not show any `(VECTOR)` legs, just from RNGRR direct to RBV. So my new guess is that they're just illustrating the radial, but don't want you to fly it past RNGRR.

Removed the heading after RNGRR.

image

image

Comment (Fechulo) [reply assets/airports/kjfk.json]: Works for me! 👍

**Pull request merged**